pipeline_steps/common.groovy

```diff
@@ -670,6 +670,13 @@ String clone_with_pr_refs(
 
 
 String clone_repo(String directory, String ssh_key, String repo, String ref, String refspec) {
+  // Need to clone/fetch with ssh@ protocol
+  if (ref.startsWith("https://")) {
+    beforeConversion = ref
+    ref = ref.replaceAll("https://", "git@")
+    ref = ref.replaceAll("github.com:", "github.com/")
```

Comment:
Are we converting from https://github.com/org/repo to git@github.com:org/repo or the other way around? Line 677 and line 676 are not matching in their from & to replacements.

Reply:
Good catch!